### PR TITLE
tests: move reset seed to fixture [1/2]

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

cleaning examples from seeding used for testing
try to add doctest fixture to preserve test's stability

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2702.org.readthedocs.build/en/2702/

### CODE REVIEW

File: src/conftest.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+from pathlib import Path
+from typing import Optional
+
+from lightning_utilities.core.imports import package_available
+
+testing = package_available("pytest") and package_available("doctest")
+
+if testing:
```
Comment (Borda) [inline src/conftest.py:8]:
```suggestion
if package_available("pytest") and package_available("doctest"):
```

File: src/conftest.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+from pathlib import Path
+from typing import Optional
+
+from lightning_utilities.core.imports import package_available
+
+if package_available("pytest") and package_available("doctest"):
```
Comment (SkafteNicki) [inline src/conftest.py:6]: Can we make this check more restrictive so it only runs during doctests? Basically asking if this code also will run on `import torchmetrics`?

Comment (Borda) [reply src/conftest.py:6]: no, it shall not as it is outside `torchmetrics` package, it is directly in `src/`

**Review by SkafteNicki — APPROVED** [review]
